newrelic/structures_newrelic_one_dashboard.go

```diff
@@ -60,8 +63,22 @@ func expandDashboardVariablesInput(variables []interface{}) []dashboards.Dashboa
 		var variable dashboards.DashboardVariableInput
 		v := val.(map[string]interface{})
 
-		if d, ok := v["default_values"]; ok && len(d.([]interface{})) > 0 {
-			variable.DefaultValues = expandVariableDefaultValues(d.([]interface{}))
+		if d, ok := v["default_values"].([]interface{}); ok && d != nil {
+			// Check if the slice is empty or contains only nil elements as we are receiving d as [<nil>] when an empty string is given
+			hasNonNil := false
+			for _, item := range d {
+				if item != nil {
+					hasNonNil = true
+					break
+				}
+			}
+			if len(d) > 0 && hasNonNil {
+				variable.DefaultValues = expandVariableDefaultValues(d)
+			} else {
+				return []dashboards.DashboardVariableInput{}, fmt.Errorf("default_values is an empty slice or contains only nil values")
+			}
+		} else {
+			return []dashboards.DashboardVariableInput{}, fmt.Errorf("default_values is either nil, not a slice, or an empty slice")
```

Comment:
@vinay-newrelic looks like there are a couple of issues with the code changes made.

https://github.com/user-attachments/assets/6ba46281-d1a7-4f09-91b4-4dfa42454696

- in the first test, if `default_values = [""]`, the error is thrown, which is ✅ 
- in the second test, if `default_values = []`, the error is thrown, which is ✅
- in the third test, if `default_values` does not exist in the `variable{}` block, the error is thrown, which is not supposed to happen, as `default_values` is an optional argument in `variable{}`. This would amount to a breaking change; so we need to make sure this does not happen because of the changes.
- in the fourth test, if `default_values = ["", "*"]` (the `*` can be any other non empty string), a panic is still thrown, so we might need to change the logic accordingly

Please take a look and add changes to make sure these cases are covered :)

Reply:
@pranav-new-relic I agree upon the third test case and  removing the mandatory check on default values would solve the issue but regarding the fourth test case ...will add **`_has atleast one nill_`** check instead of **_non nil check_** which is being done currently